src/contrib/cluster/Akka.Cluster.Tools.Tests.MultiNode/ClusterClient/ClusterClientSpec.cs

```diff
@@ -141,194 +146,268 @@ private RoleName GetRoleName(Address address)
                     _remainingServerRoleNames.Remove(receptionistRoleName);
 
-                    Within(Remaining - TimeSpan.FromSeconds(3), () =>
+                    Within(Remaining - 3.Seconds(), () =>
                     {
                         AwaitAssert(() =>
                         {
-                            c.Tell(new Client.ClusterClient.Send("/user/service2", "hi again", localAffinity: true));
-                            ExpectMsg("hi again-ack", TimeSpan.FromSeconds(1));
+                            c.Tell(new ClusterClient.Send("/user/service2", "hi again", localAffinity: true));
+                            ExpectMsg<ClusterClientSpecConfig.Reply>(1.Seconds()).Msg.Should().Be("hi again-ack");
                         });
                     });
                     Sys.Stop(c);
-                }, _client);
+                }, _config.Client);
                 EnterBarrier("verified-3");
 
-                ReceiveWhile(TimeSpan.FromSeconds(2), msg =>
+                ReceiveWhile(2.Seconds(), msg =>
                 {
                     if (msg.Equals("hi again")) return msg;
-                    else throw new Exception("unexpected message: " + msg);
+                    else throw new Exception("Unexpected message: " + msg);
                 });
                 EnterBarrier("after-4");
             });
         }
 
-        //[MultiNodeFact(Skip = "TODO")]
-        public void ClusterClient_should_reestablish_connection_to_receptionist_after_partition()
+        public void ClusterClient_must_reestablish_connection_to_receptionist_after_partition()
         {
-            ClusterClient_should_reestablish_connection_to_another_receptionist_when_server_is_shutdown();
-
-            Within(TimeSpan.FromSeconds(30), () =>
+            Within(30.Seconds(), () =>
             {
                 RunOn(() =>
                 {
-                    var c = Sys.ActorOf(Client.ClusterClient.Props(ClusterClientSettings.Create(Sys).WithInitialContacts(InitialContacts)), "client3");
-                    c.Tell(new Client.ClusterClient.Send("/user/service2", "bonjour2", localAffinity: true));
-                    ExpectMsg("bonjour2-ack");
-                    var lastSenderAddress = LastSender.Path.Address;
+                    var c = Sys.ActorOf(ClusterClient.Props(ClusterClientSettings.Create(Sys).WithInitialContacts(InitialContacts)), "client3");
+                    c.Tell(new ClusterClient.Send("/user/service2", "bonjour2", localAffinity: true));
+                    var reply = ExpectMsg<ClusterClientSpecConfig.Reply>();
+                    reply.Msg.Should().Be("bonjour2-ack");
 
-                    var receptionistRoleName = RoleName(lastSenderAddress);
-                    if (receptionistRoleName == null) throw new Exception("Unexpected missing role name: " + lastSenderAddress);
+                    RoleName receptionistRoleName;
+                    if (GetRoleName(reply.Node) != null)
```

Comment:
It fails here with `TaskCanceledException`. Also there is an exception in Controller
```
[ERROR][03.08.2016 17:55:04][Thread 0023][akka://MultiNodeClusterSpec/user/controller] The given key was not present in the dictionary.
Cause: System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.Collections.Immutable.ImmutableDictionary`2.get_Item(TKey key)
   at Akka.Remote.TestKit.Controller.OnReceive(Object message) in C:\Projects\akka.net\src\core\Akka.Remote.TestKit\Controller.cs:line 370
   at Akka.Actor.UntypedActor.Receive(Object message) in C:\Projects\akka.net\src\core\Akka\Actor\UntypedActor.cs:line 21
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message) in C:\Projects\akka.net\src\core\Akka\Actor\ActorBase.cs:line 154
   at Akka.Actor.ActorCell.ReceiveMessage(Object message) in C:\Projects\akka.net\src\core\Akka\Actor\ActorCell.DefaultMessages.cs:line 168
   at Akka.Actor.ActorCell.Invoke(Envelope envelope) in C:\Projects\akka.net\src\core\Akka\Actor\ActorCell.DefaultMessages.cs:line 77
```